moralization/inout.py

```diff
@@ -37,16 +39,29 @@ def get_input_file(filename: str) -> object:
         return data
 
     @staticmethod
-    def get_input_dir(dir: str) -> dict:
+    def get_input_dir(dir: str, use_custom_ts=False) -> dict:
         "Get a list of input files from a given directory. Currently only xmi files."
         ### load multiple files into a list of dictionaries
-        ts = InputOutput.read_typesystem()
         dir_path = pathlib.Path(dir)
         if not dir_path.is_dir():
             raise RuntimeError(f"Path {dir_path} does not exist")
         data_files = dir_path.glob("*.xmi")
         if not data_files:
             raise RuntimeError(f"No input files found in {dir_path}")
+
+        if use_custom_ts is False:
+            ts = InputOutput.read_typesystem()
+        else:
+            ts_file = list(dir_path.glob("TypeSystem.xml"))
+            if len(ts_file) > 1:
+                raise Warning("Multiple typesystems found. Please provide only one.")
+            elif len(ts_file) == 0:
+                raise FileNotFoundError(
+                    f"Trying to find custom typesystem, but no 'TypeSystem.xml' found in {dir_path}"
+                )
+            else:
+                ts = InputOutput.read_typesystem(ts_file[0])
+
```

Comment:
I would have a single `read_typesystem` call after the conditional logic for clarity, instead of having two inside nested conditionals, e.g. (didn't test this code)
```suggestion
        ts_file = None
        if use_custom_ts:
            ts_files = list(dir_path.glob("TypeSystem.xml"))
            if len(ts_files) > 1:
                raise Warning("Multiple typesystems found. Please provide only one.")
            elif len(ts_files) == 0:
                raise FileNotFoundError(
                    f"Trying to find custom typesystem, but no 'TypeSystem.xml' found in {dir_path}"
                )
            ts_file = ts_files[0]
        ts = InputOutput.read_typesystem(ts_file)

```